### PR TITLE
Feature/carousel and style

### PR DESCRIPTION
**Pull request by Justicesvr**
## Description

Please include a summary of the change or which issue is fixed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [x] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [x] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [x] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [ ] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by No3ml3 — DISMISSED** [review]

**Review by Cardamced — DISMISSED** [review]
Ok.

File: packages/frontend/web/public/header-profile.png
Comment (YDrogen) [inline packages/frontend/web/public/header-profile.png:1]: Pourquoi l'image est aussi grande ? Pourquoi autant de vide autour ?

File: packages/frontend/web/src/components/cocktail-detail/Carousel.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import { ChevronLeft, ChevronRight } from 'lucide-react';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+
+import type { Cocktail } from '@app/types';
+
+const modulo = (number: number, modulus: number) => {
+  return ((number % modulus) + modulus) % modulus;
+};
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/Carousel.tsx:9]: Cette fonction beaucoup trop complexe ne sert à rien :(

File: packages/frontend/web/src/components/cocktail-detail/Carousel.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+export default function Carousel({
+  similarCocktails,
+}: {
+  readonly similarCocktails: Cocktail[];
+}) {
+  const [index, setIndex] = useState(0);
+
+  const handlePrevious = () => {
+    setIndex((currentIndex) =>
+      modulo(currentIndex - 1, similarCocktails.length),
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/Carousel.tsx:37]:
```suggestion
      (currentIndex - 1) % similarCocktails.length,
```
Juste ça, ça suffit `number % modulus`

**Review by Cardamced — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/frontend/web/src/components/profile-page/CommentsSection.tsx [hunk 3]
```diff
@@ -13,65 +9,20 @@ interface CommentsSectionProfileProps {
...
-                  color='#0E0F0F'
-                  className='my-auto me-3 h-7 w-7 cursor-pointer stroke-[4px]'
-                />
-              </button>
-            )}
-            <h2 className='pe-2'>{`review`}</h2>
-          </div>
+        <div className='mb-20 ms-8 mt-5 flex grid h-full w-full items-center justify-center gap-5 md:grid-cols-2 lg:grid-cols-3'>
+          {comments === null ? (
+            '"No comments yet"'
```
Comment (YDrogen) [inline packages/frontend/web/src/components/profile-page/CommentsSection.tsx:20]: C'est normal les quotes dans des quotes ?

File: packages/frontend/web/src/pages/ProfilePage.tsx [hunk 4]
```diff
@@ -64,21 +64,27 @@ export default function ProfilePage() {
...
-            <div className='h-[400px] md:my-5 md:grid md:grid-cols-2 md:gap-5 md:gap-y-16 lg:grid-cols-3 xl:grid-cols-4'>
-              {user.cocktails === null ? (
-                'there is no cocktails for the moment'
-              ) : (
+
+        {user.cocktails === null ? (
+          <div className='bg-light border-dark relative top-[-40px] z-10 w-full border-y-[6px] pe-1 sm:top-[-60px] md:w-[50%] md:border-[6px]'>
+            <p className='my-12 flex justify-center'>
+              {' '}
+              {'"There is no cocktails for the moment"'}
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/ProfilePage.tsx:75]: même question

File: packages/frontend/web/src/pages/ProfilePage.tsx [hunk 5]
```diff
@@ -64,21 +64,27 @@ export default function ProfilePage() {
...
-          <div className='bg-light scrollbar-bigger-rounded h-full w-full overflow-y-hidden md:overflow-x-hidden md:overflow-y-scroll'>
-            <div className='h-[400px] md:my-5 md:grid md:grid-cols-2 md:gap-5 md:gap-y-16 lg:grid-cols-3 xl:grid-cols-4'>
-              {user.cocktails === null ? (
-                'there is no cocktails for the moment'
-              ) : (
+
+        {user.cocktails === null ? (
+          <div className='bg-light border-dark relative top-[-40px] z-10 w-full border-y-[6px] pe-1 sm:top-[-60px] md:w-[50%] md:border-[6px]'>
+            <p className='my-12 flex justify-center'>
+              {' '}
```
Comment (YDrogen) [inline packages/frontend/web/src/pages/ProfilePage.tsx]: espace vide normal ?

File: packages/frontend/web/src/components/cocktail-detail/Carousel.tsx [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        type='button'
+        className='xs:-left-[5rem] absolute top-[10rem] z-10 h-10 w-10'
+        onClick={handlePrevious}
+      >
+        <ChevronLeft className='h-10 w-10 cursor-pointer' />
+      </button>
+      {similarCocktails.map((cocktail, itemIndex) => (
+        <div
+          key={cocktail.id}
+          className={` ${getClassName(itemIndex)} ${
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/Carousel.tsx]: espace vide

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/public/header-profile.png
Comment (Justicesvr) [reply packages/frontend/web/public/header-profile.png:1]: ba je ne sais pas c'est comme ca je l'ai exporté sur figma

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

File: packages/backend/api/uploads/42eda93b-68a7-44b1-9c83-688cf3e71acb_cocktail-placeholder.png.png
Comment (YDrogen) [inline packages/backend/api/uploads/42eda93b-68a7-44b1-9c83-688cf3e71acb_cocktail-placeholder.png.png:1]: 1) Le contenu de `uploads` ne doit pas être commit (à ajouter au gitignore)
2) Pourquoi `.png.png` ? Y'a un soucis quelque part je pense

File: packages/frontend/web/src/components/cocktail-detail/StarRating.tsx [hunk 7]
```diff
@@ -1,33 +1,16 @@
...
-  score: number;
-  user_id: number;
-  cocktail_id: number;
-};
-
-export default function StarRating({ ratings }: StarRatingProps) {
-  return (
-    <div>
-      {ratings?.map((rating) => (
+export default function StarRating({ rating }: RatingProps) {
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/StarRating.tsx]: C'est une bonne pratique d'appeler le typage des props comme le nom du component avec Props en suffixe
`StarRatingProps`

File: packages/frontend/web/src/components/cocktail-detail/SimilarCocktail.tsx [hunk 8]
```diff
@@ -4,6 +4,8 @@ import { Link } from 'react-router-dom';
 
 import type { Cocktail } from '@app/types';
 
+import Carrousel from './Carousel';
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/SimilarCocktail.tsx]: Faut d'orthographe

File: packages/backend/api/src/comment.ts [hunk 9]
```diff
@@ -15,55 +25,94 @@ commentRouter.get('/', async (req, res) => {
...
     console.error(error);
     res.status(500).send('Internal Server Error');
   }
 });
 
+declare module 'express-serve-static-core' {
+  interface Request {
+    userId?: number;
+  }
+}
```
Comment (YDrogen) [inline packages/backend/api/src/comment.ts]: Sert à rien ?

File: packages/backend/api/src/comment.ts [hunk 10]
```diff
@@ -15,55 +25,94 @@ commentRouter.get('/', async (req, res) => {
...
+      rating 
+      ON 
+      comment.id = rating.id 
+      AND comment.cocktail_id = rating.cocktail_id
+      JOIN
+      user
+      ON
+      comment.user_id = user.id
+      WHERE 
+      comment.cocktail_id = ${id};
```
Comment (YDrogen) [inline packages/backend/api/src/comment.ts]: pourquoi
autant
de
sauts
de
lignes
?

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by jaalazard — DISMISSED** [review]

**Review by Cardamced — DISMISSED** [review]

File: packages/frontend/web/src/components/cocktail-detail/StarRating.tsx [hunk 11]
```diff
@@ -1,33 +1,16 @@
-export interface StarRatingProps {
-  readonly ratings: StarRating[] | undefined;
+interface StarRatingProps {
+  readonly rating: number | null | undefined;
```
Comment (YDrogen) [inline packages/frontend/web/src/components/cocktail-detail/StarRating.tsx]: Pourquoi `null` ?

**Review by YDrogen — CHANGES_REQUESTED** [review]
Il ne faut pas oublier d'ajouter le dossier uploads dans le gitignore

**Review by No3ml3 — DISMISSED** [review]

**Review by jaalazard — DISMISSED** [review]

**Review by jaalazard — APPROVED** [review]
Oui

**Review by Cardamced — APPROVED** [review]

**Review by YDrogen — APPROVED** [review]

**Pull request merged**